salt/modules/ps.py

```diff
@@ -632,3 +632,73 @@ def get_users():
                                    'started': started, 'host': rec[5]})
         except ImportError:
             return False
+
+
+def lsof(name):
+    '''
+    Retrieve the lsof informations of the given process name.
+
+    CLI Example:
+
+    .. code-block:: bash
+        
+        salt '*' ps.lsof apache2
+    '''
+    sanitize_name = str(name)
+    lsof_infos =  __salt__['cmd.run']("lsof -c " + sanitize_name)
+    ret = []
+    ret.extend([sanitize_name, lsof_infos])
+    return ret
+
+
+def netstat(name):
+    '''
+    Retrieve the netstat informations of the given process name.
+
+    CLI Example:
+
+    .. code-block:: bash
+        
+        salt '*' ps.netstat apache2
+    '''
+    sanitize_name = str(name)
+    netstat_infos =  __salt__['cmd.run']("netstat -nap")
+    found_infos = []
+    ret = []
+    for info in netstat_infos.splitlines():
+        if info.find(sanitize_name) != -1:
+            found_infos.append(info)
+    ret.extend([sanitize_name, found_infos])
+    return ret
+
+
+def psaux(name):
+    '''
+    Retrieve information corresponding to a "ps aux" filtered
+    with the given pattern. It could be just a name or a regular
+    expression (using python search from "re" module).
+
+    CLI Example:
+
+    .. code-block:: bash
+                
+        salt '*' ps.psaux www-data.+apache2
+    '''
+    import re
+    sanitize_name = str(name)
+    pattern = re.compile(sanitize_name)
+    ps_aux =  __salt__['cmd.run']("ps aux")
+    found_infos = []
+    ret = []
+    nb_lines = 0
+    for info in ps_aux.splitlines():
+        found = pattern.search(info)
+        if found is not None:
+            # remove 'salt' command from results
+            if info.find("salt") == -1:
```

Comment:
Again, I understand the intention here but this just blindly removes _all_ matches of that string. We need to ensure that we're *only* removing the process matching our PID.